web/src/components/molecules/Common/Form/AssetItem/hooks.ts

```diff
@@ -24,25 +28,42 @@ export default (
     setUploadModalVisibility(true);
   }, [setUploadModalVisibility]);
 
-  const hideUploadModal = useCallback(() => {
-    setUploadModalVisibility(false);
-    setUploading(false);
-    setFileList([]);
-  }, [setFileList, setUploadModalVisibility, setUploading]);
-
   const handleUpload = useCallback(async () => {
     setUploading(true);
-    const assets = await createAssets(fileList);
-    if (assets?.length > 0) onLink(assets[0]);
-    hideUploadModal();
-  }, [createAssets, fileList, hideUploadModal, onLink, setUploading]);
+    let assets: (Asset | undefined)[] = [];
+    let asset: Asset | undefined;
+    try {
+      switch (uploadType) {
+        case "url":
+          asset = await onAssetCreateFromUrl(uploadUrl);
+          break;
+        case "local":
+        default:
+          assets = await onAssetsCreate(fileList);
+          if (assets.length > 0) asset = assets[0];
+          break;
+      }
+      if (asset) onLink(asset);
+      hideUploadModal();
+    } catch (error) {
+      hideUploadModal();
+    }
```

Comment:
You aren't handling this error. Also, if all routes end up with hideUploadModal, put it outside of any conditional statements at the end of the callback.

Reply:
The error has been handles from these two functions `onAssetsCreate` and `onAssetCreateFromUrl`.
in this function I just want to hide the upload modal if an error happened.

if I put the `hideUploadModal` function outside the switch statement it will not run if an error happened.

Reply:
Please let me know what do you think.

Reply:
Just a couple things.
If the error is being handled in those functions, I don't think you need the try/catch, since you aren't catching anything.
Also, probably you just mistakenly wrote switch instead of try/catch above, but hideUploadModal is inside the try/catch, so by removing the try/catch you kinda solve the issue anyways :P